lib/ts/recipe/passwordless/index.ts

```diff
@@ -41,39 +41,83 @@ export default class RecipeWrapper {
         flowType: PasswordlessFlowType;
         fetchResponse: Response;
     }> {
-        return Recipe.getInstanceOrThrow().recipeImplementation.createCode({
+        const recipe: Recipe = Recipe.getInstanceOrThrow();
+        const normalisedUserContext = getNormalisedUserContext(input.userContext);
+
+        const createCodeResponse = await recipe.recipeImplementation.createCode({
             ...input,
-            userContext: getNormalisedUserContext(input.userContext),
+            userContext: normalisedUserContext,
         });
+
+        await recipe.recipeImplementation.setLoginAttemptInfo({
+            attemptInfo: {
+                deviceId: createCodeResponse.deviceId,
+                preAuthSessionId: createCodeResponse.preAuthSessionId,
+                flowType: createCodeResponse.flowType,
+            },
+            userContext: normalisedUserContext,
+        });
+
+        return createCodeResponse;
     }
 
-    static resendCode(input: {
-        deviceId: string;
-        preAuthSessionId: string;
-        userContext?: any;
-        options?: RecipeFunctionOptions;
-    }): Promise<{
+    static async resendCodeAndUpdateState(input: { userContext?: any; options?: RecipeFunctionOptions }): Promise<{
         status: "OK" | "RESTART_FLOW_ERROR";
         fetchResponse: Response;
     }> {
-        return Recipe.getInstanceOrThrow().recipeImplementation.resendCode({
+        const recipe: Recipe = Recipe.getInstanceOrThrow();
+        const normalisedUserContext = getNormalisedUserContext(input.userContext);
+
+        const previousAttemptInfo = await recipe.recipeImplementation.getLoginAttemptInfo({
+            userContext: normalisedUserContext,
+        });
+
+        if (previousAttemptInfo === undefined) {
```

Comment:
There is a race condition here that if a user has both OTP and magic link, and they have enter OTP opened on one screen and then click on the magic link to login (which clears the login attempt info), then if they go back to the top screen and click on resend, they will see an error.

In the auth-react SDK, this case of it being `undefined` is solved by showing a different screen to the user asking them to close the tab. But i would like this to be so that our users don't have to think about this edge case and we should somehow gracefully handle this?

That being said, even in the auth-react SDK, I don't think this type of check if happening (even though it should be) - maybe ask @porcellus about how this is handled in the SDK?

Reply:
Shouldn't we return a `RESTART_FLOW_ERROR` in this case? That can be done by just setting these as empty strings and then calling the API?

Reply:
After discussing this with @porcellus, I have refactored this to use empty strings if `previousAttemptInfo` to return a `RESTART_FLOW_ERROR`